tests/components/bbox/test_device_tracker.py

```diff
@@ -0,0 +1,177 @@
+"""Tests for the Bbox device tracker platform."""
+
+from unittest.mock import MagicMock, patch
+
+import pytest
+import requests
+
+from homeassistant.components.bbox.device_tracker import (
+    PLATFORM_SCHEMA,
+    BboxDeviceScanner,
+    Device,
+    get_scanner,
+)
+from homeassistant.components.device_tracker import DOMAIN as DEVICE_TRACKER_DOMAIN
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.typing import ConfigType
+
+
+@pytest.fixture
+def mock_bbox_api():
+    """Mock the pybbox API."""
+    with patch("homeassistant.components.bbox.device_tracker.pybbox") as mock_pybbox:
+        mock_box = MagicMock()
+        mock_pybbox.Bbox.return_value = mock_box
+        yield mock_box
+
+
+@pytest.fixture
+def device_tracker_config():
+    """Return device tracker configuration."""
+    return {DEVICE_TRACKER_DOMAIN: {"host": "192.168.1.254"}}
+
+
+async def test_get_scanner_success(
+    hass: HomeAssistant, device_tracker_config: ConfigType, mock_bbox_api
+) -> None:
+    """Test successful scanner initialization."""
+    mock_bbox_api.get_all_connected_devices.return_value = [
+        {
+            "macaddress": "aa:bb:cc:dd:ee:ff",
+            "hostname": "test_device",
+            "ipaddress": "192.168.1.100",
+            "active": 1,
+        }
+    ]
+
+    scanner = get_scanner(hass, device_tracker_config)
+    assert scanner is not None
+    assert scanner.success_init is True
+
+
+async def test_get_scanner_failure(
+    hass: HomeAssistant, device_tracker_config: ConfigType, mock_bbox_api
+) -> None:
+    """Test scanner initialization failure."""
+    mock_bbox_api.get_all_connected_devices.side_effect = requests.exceptions.HTTPError(
+        "Connection failed"
+    )
+    with patch.object(BboxDeviceScanner, "_update_info", return_value=False):
+        scanner = get_scanner(hass, device_tracker_config)
+        assert scanner is None
+
+
+async def test_scan_devices(
+    hass: HomeAssistant, device_tracker_config: ConfigType
+) -> None:
+    """Test scanning for devices."""
+    with patch.object(BboxDeviceScanner, "_update_info") as mock_update:
```

Comment:
Can we try to avoid mocking individual methods, but entire classes? If we mock only methods, then when our tests run, not only our code is tested, but also the external library code.